### PR TITLE
Renames needed for customer workshop

### PR DESCRIPTION
**Pull request by m-nash**
Also fixed a few style cop things in files I edited

### CODE REVIEW

**Review by allenjzhang — APPROVED** [review]

File: azure-proto-compute/AvailabilitySet.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using azure_proto_core;
+
```
Comment (allenjzhang) [inline azure-proto-compute/AvailabilitySet.cs:2]: nit: file header

File: azure-proto-core/ArmClient.cs [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
-using System.Collections.Generic;
-using System.Linq;
-using System.Runtime.CompilerServices;
-using System.Threading;
-using System.Threading.Tasks;
-
-namespace azure_proto_core
+﻿namespace azure_proto_core
 {
+    using System;
```
Comment (allenjzhang) [inline azure-proto-core/ArmClient.cs:3]: We will need to change this to outside. Let me know if the ruleset was incorrectly set for this one.

File: azure-proto-compute/VirtualMachine.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    {
+        public VirtualMachine(ArmClientContext context, VirtualMachineData resource, ArmClientOptions options)
+            : base(context, resource.Id, options)
+        {
+            Model = resource;
+        }
+
+        public VirtualMachineData Model { get; private set; }
+    }
+}
```
Comment (bquantump) [inline azure-proto-compute/VirtualMachine.cs:15]: Why is this file showing a new if this is a just a re-name PR?

File: azure-proto-core/SubscriptionOperations.cs [hunk 4]
```diff
@@ -33,27 +32,29 @@ internal SubscriptionOperations(ArmClientContext context, Resource subscription,
...
             new ResourcesManagementClient(uri, Guid.NewGuid().ToString(), cred, ArmClientOptions.Convert<ResourcesManagementClientOptions>(ClientOptions))).Subscriptions;
-        internal ResourceGroupsOperations RgOperations => GetClient<ResourcesManagementClient>((uri, cred) => 
+
+        internal ResourceGroupsOperations RgOperations => GetClient<ResourcesManagementClient>((uri, cred) =>
             new ResourcesManagementClient(uri, Id.Subscription, cred, ArmClientOptions.Convert<ResourcesManagementClientOptions>(ClientOptions))).ResourceGroups;
 
         public Pageable<ResourceGroupOperations> ListResourceGroups(CancellationToken cancellationToken = default)
         {
-            return new PhWrappingPageable<ResourceGroup, ResourceGroupOperations>(
+            return new PhWrappingPageable<Azure.ResourceManager.Resources.Models.ResourceGroup, ResourceGroupOperations>(
```
Comment (bquantump) [inline azure-proto-core/SubscriptionOperations.cs:44]: Why did we have to full specify this now? Azure.ResourceManager.Resources.Models.ResourceGroup from ResourceGroup.

File: client/Scenarios/ShutdownVmsByTag.cs [hunk 5]
```diff
@@ -18,7 +18,7 @@ public override void Execute()
             Random rand = new Random(Environment.TickCount);
             foreach (var generic in rg.VirtualMachines().ListByName(Environment.UserName))
             {
-                var vm = new VirtualMachineOperations(generic);
+                var vm = VirtualMachineOperations.FromGeneric(generic);
```
Comment (bquantump) [inline client/Scenarios/ShutdownVmsByTag.cs:21]: We decided on the FromGeneric pattern then?

File: azure-proto-compute/VirtualMachine.cs (hunk 3 above)
Comment (bquantump) [reply azure-proto-compute/VirtualMachine.cs:15]: Looks like github

File: azure-proto-core/SubscriptionOperations.cs (hunk 4 above)
Comment (m-nash) [reply azure-proto-core/SubscriptionOperations.cs:44]: In order to differentiate between azure_proto_core.ResourceGroup and Azure.ResourceManager.Resources.Models.ResourceGroup

File: client/Scenarios/ShutdownVmsByTag.cs (hunk 5 above)
Comment (m-nash) [reply client/Scenarios/ShutdownVmsByTag.cs:21]: No, this is the straw man that allows us to maintain all constructors internal for now.  This is still consideration

File: azure-proto-compute/AvailabilitySet.cs [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+{
+    public class AvailabilitySet : AvailabilitySetOperations
+    {
+        public AvailabilitySet(ArmClientContext context, AvailabilitySetData resource, ArmClientOptions options)
+            : base(context, resource.Id, options)
+        {
+            Model = resource;
+        }
+
+        public AvailabilitySetData Model { get; private set; }
```
Comment (markcowl) [inline azure-proto-compute/AvailabilitySet.cs:13]: Should the Property name be 'Data' if the type suffix is 'Data'?

**Review by bquantump — APPROVED** [review]

File: azure-proto-compute/AvailabilitySetContainer.cs [hunk 7]
```diff
@@ -12,80 +12,86 @@ namespace azure_proto_compute
     /// <summary>
     /// Operatiosn class for Availability Set Contaienrs (resource groups)
     /// </summary>
-    public class AvailabilitySetContainer : ResourceContainerOperations<XAvailabilitySet, PhAvailabilitySet>
+    public class AvailabilitySetContainer : ResourceContainerOperations<AvailabilitySet, AvailabilitySetData>
     {
-        public AvailabilitySetContainer(ArmClientContext context, PhResourceGroup resourceGroup, ArmClientOptions clientOptions) : base(context, resourceGroup, clientOptions) { }
+        internal AvailabilitySetContainer(ArmClientContext context, ResourceGroupData resourceGroup, ArmClientOptions clientOptions)
```
Comment (markcowl) [inline azure-proto-compute/AvailabilitySetContainer.cs:17]: I wonder if, given the return types from most of our methods, this should have constructors that accept ResourceGroup and ResourceGroupOperations.  Also, I assume we are combining client context and client options in a subsequent PR

File: azure-proto-compute/VirtualMachineOperations.cs [hunk 8]
```diff
@@ -11,14 +11,25 @@ namespace azure_proto_compute
...
 
-        internal VirtualMachineOperations(ArmClientContext context, ResourceIdentifier id, ArmClientOptions clientOptions) : base(context, id, clientOptions) { }
+        internal VirtualMachineOperations(ArmClientContext context, ResourceIdentifier id, ArmClientOptions clientOptions)
+            : base(context, id, clientOptions)
+        {
+        }
 
         public override ResourceType ResourceType => "Microsoft.Compute/virtualMachines";
 
+        public static VirtualMachineOperations FromGeneric(ArmResourceOperations genericOperations)
```
Comment (markcowl) [inline azure-proto-compute/VirtualMachineOperations.cs:28]: CreateFromGeneric?

**Review by markcowl — CHANGES_REQUESTED** [review]

File: azure-proto-compute/VirtualMachineOperations.cs (hunk 8 above)
Comment (m-nash) [reply azure-proto-compute/VirtualMachineOperations.cs:28]: Following the convention from System.DateTime and other types that have .FromBinary(...), . FromTime(...) etc

File: azure-proto-compute/AvailabilitySetContainer.cs (hunk 7 above)
Comment (m-nash) [reply azure-proto-compute/AvailabilitySetContainer.cs:17]: context / options is in a future work item.

Another future item is to remove the constructors on Operations classes that take the full resource.  The idea is if you have a full resource you construct a VirtualMachine, if you have just a ResourceIdentifier you construct a VirtualMachineOperations (both internally of course)

**Review by markcowl — APPROVED** [review]

**Pull request merged**